scripts/parse_fallback_dirs.py

```diff
@@ -0,0 +1,24 @@
+import urllib
+import csv
+
+link = "https://gitweb.torproject.org/tor.git/plain/src/app/config/fallback_dirs.inc"
+f = urllib.request.urlopen(link)
+myfile = f.read()
+
+intext = str(myfile)
+strings = intext.split("/* ===== */")[2:-1]
+lines = []
+
+for s in strings:
+    line = []
+    line.append(s.split(" ")[0].split("\\n\"")[1].split(":")[0]) 
+    line.append(s.split("orport=")[1].split(" ")[0])
+    line.append(s.split("nickname=")[1].split(" ")[0])
+    line.append("or_port")
+    lines.append(line)
+ 
+with open('tor-bridges-ip-port-updated.csv', mode='w') as csv_file:
+    csv_writer = csv.writer(csv_file, delimiter=',')
+
+    for l in lines:
+        csv_writer.writerow(l)
```

Comment:
Thanks for also writing a script to parse them!

I guess this does not actually write directly the csv file, because the csv file contains some extra IPs and addresses from other parts of the tor source code.

How did you go abouts merging what is the output of this script with the content of the existing CSV?

It would be great if we had a way to entirely generate the csv file from a script that we can schedule to run on a periodic basis.

Reply:
As far as I understand, there are 4 potential sources for hard-coded relays:
* Tor fallback directory mirrors: https://gitweb.torproject.org/tor.git/tree/src/app/config/fallback_dirs.inc
* Tor authorities: https://gitweb.torproject.org/tor.git/tree/src/app/config/auth_dirs.inc
* Tor Browser (Desktop) Bridges: these are kept in a JS preferences file, I don't know where it is
* Tor Browser for Android Bridges: I'm not sure if they use a different set of bridges

Reply:
>     * Tor Browser (Desktop) Bridges: these are kept in a JS preferences file, I don't know where it is

FWIW, they are available here: https://gitweb.torproject.org/builders/tor-browser-build.git/tree/projects/tor-browser/Bundle-Data/PTConfigs/bridge_prefs.js

>     * Tor Browser for Android Bridges: I'm not sure if they use a different set of bridges

The list is identical to desktop, and available here: https://gitweb.torproject.org/tor-android-service.git/tree/service/src/main/assets/common/bridges.txt

More broadly, we maintain all information related to default bridges here: https://trac.torproject.org/projects/tor/wiki/doc/TorBrowser/DefaultBridges